.github/workflows/eval.yml

```diff
@@ -124,18 +153,93 @@ jobs:
       - name: Combine all output paths and eval stats
         run: |
           nix-build nixpkgs/ci -A eval.combine \
-            --arg resultsDir ./intermediate
+            --arg resultsDir ./intermediate \
+            -o prResult
 
       - name: Upload the combined results
         uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
         with:
           name: result
-          path: result/*
+          path: prResult/*
 
+      - name: Get base run id
+        if: needs.attrs.outputs.baseSha
+        id: baseRunId
+        run: |
+          # Get the latest eval.yml workflow run for the PR's base commit
+          if ! run=$(gh api --method GET /repos/"$REPOSITORY"/actions/workflows/eval.yml/runs \
+            -f head_sha="$BASE_SHA" \
+            --jq '.workflow_runs | sort_by(.run_started_at) | .[-1]') \
+            || [[ -z "$run" ]]; then
+            echo "Could not find an eval.yml workflow run for $BASE_SHA, cannot make comparison"
+            exit 0
+          fi
+          echo "Comparing against $(jq .html_url <<< "$run")"
+          runId=$(jq .id <<< "$run")
+          conclusion=$(jq -r .conclusion <<< "$run")
+
+          while [[ "$conclusion" == null ]]; do
+            echo "Workflow not done, waiting 10 seconds before checking again"
+            sleep 10
+            conclusion=$(gh api /repos/"$REPOSITORY"/actions/runs/"$runId" --jq '.conclusion')
+          done
+
+          if [[ "$conclusion" != "success" ]]; then
+            echo "Workflow was not successful, cannot make comparison"
+            exit 0
+          fi
 
-      # TODO: Run this workflow also on `push` (on at least the main development branches)
-      # Then add an extra step here that waits for the base branch (not the merge base, because that could be very different)
-      # to have completed the eval, then use
-      # gh api --method GET /repos/NixOS/nixpkgs/actions/workflows/eval.yml/runs -f head_sha=<BASE>
-      # and follow it to the artifact results, where you can then download the outpaths.json from the base branch
-      # That can then be used to compare the number of changed paths, get evaluation stats and ping appropriate reviewers
+          echo "baseRunId=$runId" >> "$GITHUB_OUTPUT"
+        env:
+          REPOSITORY: ${{ github.repository }}
+          BASE_SHA: ${{ needs.attrs.outputs.baseSha }}
+          GH_TOKEN: ${{ github.token }}
+
+      - uses: actions/download-artifact@v4
+        if: steps.baseRunId.outputs.baseRunId
+        with:
+          name: result
+          path: baseResult
+          github-token: ${{ github.token }}
+          run-id: ${{ steps.baseRunId.outputs.baseRunId }}
+
+      - name: Compare against the base branch
+        if: steps.baseRunId.outputs.baseRunId
+        run: |
+          nix-build nixpkgs/ci -A eval.compare \
```

Comment:
rather than doing the comparison with the json output  with jq  which seems to involve a bunch of processing, using the intermediate outputs seem like the number of rebuilds can be done in a one-liner with a combination of comm, awk, sort, and wc.

```
comm -13 <(awk '{ print $2 }' ./result-old/paths|sort -u) <(awk '{print $2 }' ./result-new/paths|sort -u) | wc -l
```

Reply:
Can be improved in a later pull request, I would say. Would be cool to have rebuild labels earlier than later. Nix might be also a language appropriate here because this is what the nix community can read.

Reply:
The jq script does a bunch more though: It also determines removed and added packages, it splits the platform from the attribute paths and up by kernel (linux/darwin) and turns it into appropriate labels. It can also be extended to support platform-specific labels. Though I admit it's a bit obscure, we could consider using Nix instead in the future.

Reply:
getting the added and removed paths are similar one liners.

```
comm -13 <(awk '{ print $1 }' ./result-old/paths|sort ) <(awk '{print $1 }' ./result-new/paths|sort )  > added
comm -23 <(awk '{ print $1 }' ./result-old/paths|sort ) <(awk '{print $1 }' ./result-new/paths|sort )  > removed
```
and the paths are already split up as one arch/OS per file from what I can tell and only merged when the json file is generated so getting per arch/OS data seems as easy as putting the 3 one-liners in a for-each loop to process the 4 systems.

Reply:
Thanks, though I'd rather not spend more time on this when the current code works already. Those who'd like to can always simplify it in the future.